analyzer.go

```diff
@@ -52,205 +65,236 @@ type Analyzer struct {
 	template string
 }
 
-func (a *Analyzer) processPerTargetValues(target *Target) error {
-	a.values["mod-year"] = a.values["year"]
-	a.values["mod-year-range"] = a.values["year-range"]
-	if t, err := target.ModTime(); err == nil {
-		a.values["mod-year"] = &ConstValue{RawValue: fmt.Sprint(t.Year())}
-		a.values["mod-year-range"] = &RegexpValue{RawValue: `((20\d\d\-{{mod-year}})|({{mod-year}}))`}
+func New(opts ...Option) *Analyzer {
+	var a Analyzer
+	for _, opt := range opts {
+		opt.apply(&a)
 	}
+	return &a
+}
 
-	for _, v := range a.values {
-		if err := v.Calculate(a.values); err != nil {
-			return err
-		}
+type Result struct {
+	Err        error
+	Fix        string
+	End, Start token.Pos
+}
+
+func (a *Analyzer) skipCodeGen(file *ast.File) ([]*ast.CommentGroup, []*ast.Comment) {
```

Comment:
```go
ast.IsGenerated(file)
```

Reply:
the function skipCodeGen is still needed to pass test with cgo from the golangci-linter repo. `ast.IsGenerated(file)` applied in another place, thanks.

Reply:
I think that cgo should not be supported by this linter because the header of a cgo file is special, which will lead to false positives.